packages/assets/src/loader/Loader.ts

```diff
@@ -211,4 +242,29 @@ export class Loader
     {
         return this._parsers;
     }
+
+    /** validates our parsers, right now it only checks for name conflicts but we can add more here as required! */
+    private _validateParsers()
```

Comment:
There's a function called `extensions.handleByList` at the bottom of **Assets.ts**. This is responsible for receiving the parser extensions and sorting the parsers list.

For instance, one of the problems with this is if an extension is added after load is called once. Not unimaginable, though probably rare.

We could add a custom function to handle the extensions install and update the parserHash at the same time. This seems like the better place to check for duplicates anyways.

Reply:
hmm.. could be a good case for using a Proxy?

Reply:
That could work too

Reply:
nice, yeah - feel like Proxy works well here - keeps things nice and simple!